tests/tests/test_basic_integration.py

```diff
@@ -124,8 +172,48 @@ def do_test_failed_updated_and_valid_update(
             deploy=deploy,
         )
 
+    def do_test_artifact_name(self, env, valid_image_with_mender_conf):
+        """Uploads an uncompressed artifact, and runs the whole update process
+           checking the artifact name change."""
+        devauth = DeviceAuthV2(env.auth)
+        deploy = Deployments(env.auth, devauth)
+
+        mender_device = env.device
+
+        # in the so-called clean image the artifact_info is "artifact_name=mender-image-clean\n"
+        # see: https://github.com/mendersoftware/mender-qa/pull/580/commits/a4c54e9ebf5879f0837789cbc7c74cb1102b2caa#diff-7031fcd5c934f225781febe446a34594a9a36b335c540379c0dd376a8f06ba28R263
+        # so first we expect it to be set to master, as this is what we
+        # have in the initial image, after update we expect it to change
+        expected_artifact_info = "artifact_name=mender-image-master\n"
+        artifact_info = mender_device.run(
+            "cat /etc/mender/artifact_info | sed -e '/artifact_name/q'"
+        )
+        logger.info("before update got %s" % artifact_info)
+        assert artifact_info == expected_artifact_info
+
+        mender_conf = mender_device.run("cat /etc/mender/mender.conf")
+        update_image(
+            env.device,
+            env.get_virtual_network_host_ip(),
+            install_image=valid_image_with_mender_conf(mender_conf),
+            compression_type="none",
+            devauth=devauth,
+            deploy=deploy,
+        )
+
+        # after the update we expect the artifact info to be different
+        # ("artifact_name=mender-image-clean\n")
+        # since this is what is in the embedded image decompressed
+        # in the tests/run.sh
+        expected_artifact_info = "artifact_name=mender-image-master\n"
+        artifact_info = mender_device.run(
+            "cat /etc/mender/artifact_info | sed -e '/artifact_name/q'"
+        )
+        logger.info("after update got %s" % artifact_info)
+        assert artifact_info != expected_artifact_info
```

Comment:
Same here.

Reply:
same here. I am a bit slow on the uptake. but after discussing it is fixed.